lib/purescript-cst/src/Language/PureScript/CST/Utils.hs

```diff
@@ -19,6 +19,30 @@ import Language.PureScript.CST.Types
 import qualified Language.PureScript.Names as N
 import Language.PureScript.PSString (PSString, mkString)
 
+newtype QualifiedProperName =
```

Comment:
I think these types could do with some comments. How about:

> A newtype for a qualified proper name whose ProperNameType has not yet been determined. This is a workaround for Happy's limited support for polymorphism; it is used inside the parser to allow us to write just one parser for qualified proper names which can be used for all of the different ProperNameTypes (via a call to `getQualifiedProperName`).

Reply:
Sure. Do you want me to repeat this comment for each `newtype` I introduced?

Reply:
Yes please

Reply:
Sorry to nitpick but could you please add a pipe character here so that these comments are marked as doc-comments and appear in the Haddocks?